packages/plugin-legacy/src/index.ts

```diff
@@ -309,9 +315,15 @@ function viteLegacyPlugin(options: Options = {}): Plugin[] {
       const { rollupOptions } = config.build
       const { output } = rollupOptions
       if (Array.isArray(output)) {
-        rollupOptions.output = [...output.map(createLegacyOutput), ...output]
+        rollupOptions.output = [
+          ...output.map(createLegacyOutput),
+          ...(genModern ? output : [])
+        ]
       } else {
-        rollupOptions.output = [createLegacyOutput(output), output || {}]
+        rollupOptions.output = [
+          createLegacyOutput(output),
+          ...(genModern ? [output || {}] : [])
+        ]
```

Comment:
318~326 will cause all css to be lost

Reply:
@lsdsjy When I tried to build with the latest stable version of plugin-legacy, I can't find the inline css in the output legacy js:
https://stackblitz.com/edit/vitejs-vite-nlalet?file=dist/assets/index-legacy.a5dbbf13.js

Likewise, I'm using your fork in another project of mine and the styles are completely lost.

Reply:
It's a conflict with https://github.com/vitejs/vite/pull/9761, which is a change of the vite core package, and my vite is not new enough  to discover this during developing. 😅 Let me see how to work with it.

Reply:
@wspl Hi, I've made some changes to make it compatible with #9761, and verified with the latest version of vite on my local environment. Could you please try it again?

Reply:
when will this merge? really need this @wspl

Reply:
https://www.npmjs.com/package/vite-plugin-legacy-no-module
I released a package based on this PR, hope it will help you. @xiechao1211